src/Mappers/InventoryMapper.php

```diff
@@ -306,20 +347,21 @@ static function getSupplierPartNumberFromVariation($variationData, $itemMappingM
 
     $supplierPartNumber = null;
 
-    $mainVariationId = $variationData['id'];
-    $variationNumber = $variationData['number'];
+    $variationNumber = $variationData[self::VARIATION_COL_NUMBER];
 
     $supplierPartNumber = null;
 
     switch ($itemMappingMethod) {
       case AbstractConfigHelper::ITEM_MAPPING_SKU:
-        if (array_key_exists(self::VARIATION_SKUS, $variationData) && !empty($variationData[self::VARIATION_SKUS])) {
-          $supplierPartNumber = $variationData[self::VARIATION_SKUS][0]['sku'];
+        if (array_key_exists(self::VARIATION_COL_SKUS, $variationData) && !empty($variationData[self::VARIATION_COL_SKUS])) {
+          // FIXME: should not blindly use the first SKU - only should use the one assigned to Wayfair!
+          $supplierPartNumber = $variationData[self::VARIATION_COL_SKUS][0]['sku'];
         }
         break;
       case AbstractConfigHelper::ITEM_MAPPING_EAN:
-        if (array_key_exists(self::VARIATION_BARCODES, $variationData) && !empty($variationData[self::VARIATION_BARCODES])) {
-          $supplierPartNumber = $variationData[self::VARIATION_BARCODES][0]['code'];
+        if (array_key_exists(self::VARIATION_COL_BARCODES, $variationData) && !empty($variationData[self::VARIATION_COL_BARCODES])) {
+          // TODO: find a way to avoid blindly using first barcode.
```

Comment:
What are the consequences of this and do we need to track this as a separate issue?

Reply:
I may have created a ticket for this, but not an issue, as there are too many unknowns:

1. Is anyone using `barcode` for mapping mode?
2. If 1, is anyone putting more than one barcode onto their Variations?
3. If 2, is there any practical way to identify which barcode is desired for using with Wayfair?